items:
- uid: System.Runtime.InteropServices.MarshalDirectiveException
  id: MarshalDirectiveException
  parent: System.Runtime.InteropServices
  children:
  - System.Runtime.InteropServices.MarshalDirectiveException.#ctor
  - System.Runtime.InteropServices.MarshalDirectiveException.#ctor(System.String)
  - System.Runtime.InteropServices.MarshalDirectiveException.#ctor(System.String,System.Exception)
  name: MarshalDirectiveException
  fullName: System.Runtime.InteropServices.MarshalDirectiveException
  type: Class
  source:
    remote: &o0
      path: src/mscorlib/src/System/Runtime/InteropServices/MarshalDirectiveException.cs
      branch: master
      repo: https://github.com/dotnet/coreclr.git
    path: src/mscorlib/src/System/Runtime/InteropServices/MarshalDirectiveException.cs
    startLine: 18
  assemblies:
  - System.Runtime.InteropServices.PInvoke
  namespace: System.Runtime.InteropServices
  summary: The exception that is thrown by the marshaler when it encounters a <xref href="System.Runtime.InteropServices.MarshalAsAttribute" data-throw-if-not-resolved="false"></xref> it does not support.
  syntax:
    content: 'public class MarshalDirectiveException : Exception'
    content.vb: Public Class MarshalDirectiveException Inherits Exception
  inheritance:
  - System.Object
- uid: System.Runtime.InteropServices.MarshalDirectiveException.#ctor
  id: '#ctor'
  parent: System.Runtime.InteropServices.MarshalDirectiveException
  name: MarshalDirectiveException()
  fullName: System.Runtime.InteropServices.MarshalDirectiveException.MarshalDirectiveException()
  type: Constructor
  source:
    remote: *o0
    path: src/mscorlib/src/System/Runtime/InteropServices/MarshalDirectiveException.cs
    startLine: 21
  assemblies:
  - System.Runtime.InteropServices.PInvoke
  namespace: System.Runtime.InteropServices
  summary: Initializes a new instance of the MarshalDirectiveException class with default properties.
  syntax:
    content: public MarshalDirectiveException()
    content.vb: Public Sub New
- uid: System.Runtime.InteropServices.MarshalDirectiveException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Runtime.InteropServices.MarshalDirectiveException
  name: MarshalDirectiveException(String)
  fullName: System.Runtime.InteropServices.MarshalDirectiveException.MarshalDirectiveException(System.String)
  type: Constructor
  source:
    remote: *o0
    path: src/mscorlib/src/System/Runtime/InteropServices/MarshalDirectiveException.cs
    startLine: 26
  assemblies:
  - System.Runtime.InteropServices.PInvoke
  namespace: System.Runtime.InteropServices
  summary: Initializes a new instance of the MarshalDirectiveException class with a specified error message.
  syntax:
    content: public MarshalDirectiveException(string message)
    content.vb: Public Sub New(message As String)
    parameters:
    - id: message
      type: System.String
      description: The error message that specifies the reason for the exception.
- uid: System.Runtime.InteropServices.MarshalDirectiveException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Runtime.InteropServices.MarshalDirectiveException
  name: MarshalDirectiveException(String, Exception)
  fullName: System.Runtime.InteropServices.MarshalDirectiveException.MarshalDirectiveException(System.String, System.Exception)
  type: Constructor
  source:
    remote: *o0
    path: src/mscorlib/src/System/Runtime/InteropServices/MarshalDirectiveException.cs
    startLine: 31
  assemblies:
  - System.Runtime.InteropServices.PInvoke
  namespace: System.Runtime.InteropServices
  summary: Initializes a new instance of the <xref href="System.Runtime.InteropServices.MarshalDirectiveException" data-throw-if-not-resolved="false"></xref> class with a specified error message and a reference to the inner exception that is the cause of this exception.
  syntax:
    content: public MarshalDirectiveException(string message, Exception inner)
    content.vb: Public Sub New(message As String, inner As Exception)
    parameters:
    - id: message
      type: System.String
      description: The error message that explains the reason for the exception.
    - id: inner
      type: System.Exception
      description: The exception that is the cause of the current exception. If the *inner* parameter is not null, the current exception is raised in a catch block that handles the inner exception.
references:
- uid: System.Runtime.InteropServices
  isExternal: false
  name: System.Runtime.InteropServices
  fullName: System.Runtime.InteropServices
- uid: System.Object
  parent: System
  isExternal: false
  name: Object
  fullName: System.Object
- uid: System
  isExternal: false
  name: System
  fullName: System
- uid: System.String
  parent: System
  isExternal: false
  name: String
  fullName: System.String
- uid: System.Exception
  parent: System
  isExternal: false
  name: Exception
  fullName: System.Exception
